File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        char[] chars = s.toCharArray();
        Stack<Character> stack = new Stack<>();
        int max = 0;
        for (char c : chars) {
            if (c == '(') {
                stack.add(c);
            } else if (c == ')') {
                stack.pop();
            }
            max = Math.max(max, stack.size());
        }
        return max;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.kt
class Solution {
    fun maxDepth(s: String): Int {
        var maxD = 0
        var currD = 0
        s.forEach {
            if (it == '(') {
                currD++
                maxD = Math.max(maxD, currD)
            } else if (it == ')') {
                currD--
            }
        }
        
        return maxD
    }
}